extensions-core/main/runtime/src/main/java/org/apache/camel/quarkus/main/CamelMain.java

```diff
@@ -77,6 +83,9 @@ protected void doStart() throws Exception {
     @Override
     protected void doStop() throws Exception {
         super.doStop();
+        /* Wait till the Camel shutdown is finished in its separate thread and proceed with subsequent Quarkus shutdown
+         * tasks only after that */
+        getShutdownStrategy().await(shutdownTimeoutMs, TimeUnit.MILLISECONDS);
```

Comment:
shouldn't be better to put this in `CamelMainRuntime::stop` metod ? here you are waiting to the shut-down before context.stop() is invoked

Reply:
> @ppalaga shouldn't be better to put this `await` in `CamelMainRuntime::stop` metod ? here you are waiting to the shut-down before context.stop() is invoked

Hm... if we do it that way, context.stop() will happen in parallel with CamelMain.beforeStop(), CamelMain.stop(), and CamelMain.afterStop(). Would that be the right thing to do?